### PR TITLE
Remove integer suffixes where the types in compiled code are identical.

### PR DESCRIPTION
**Pull request by eddyb**
Automatic has-same-types testing methodology can be found in #22501.
Because most of them don't work with `--pretty=typed`, compile-fail tests were manually audited.

r? @aturon

### CODE REVIEW

**Comment by alexcrichton** [comment]
Nice! From what I can tell there are no substantial changes other than removing integer suffixes, so r=me whenever this is ready to go.

**Comment by pnkfelix** [comment]
@eddyb it looks to me like this and PR #23002 will be racing for who can update the files first... we'll see if bors rejects me

File: src/test/compile-fail/issue-17651.rs [hunk 1]
```diff
@@ -13,6 +13,6 @@
 
 fn main() {
     // FIXME (#22405): Replace `Box::new` with `box` here when/if possible.
-    (|| Box::new(*[0_usize].as_slice()))();
-    //~^ ERROR the trait `core::marker::Sized` is not implemented for the type `[usize]`
+    (|| Box::new(*[0].as_slice()))();
+    //~^ ERROR the trait `core::marker::Sized` is not implemented for the type `[i32]`
```
Comment (pnkfelix) [inline src/test/compile-fail/issue-17651.rs]: the error message we are actually getting here is:
> error: the trait `core::marker::Sized` is not implemented for the type `[_]`

which i admit, raises a semi-interesting question about whether we are better off applying int-fallback to our *error message output* (as it seems like @eddyb assumed would be done here?)

**Comment by eddyb** [comment]
> the error message we are actually getting here is:
> > error: the trait `core::marker::Sized` is not implemented for the type `[_]`
>
> which i admit, raises a semi-interesting question about whether we are better off applying int-fallback > to our *error message output* (as it seems like @eddyb assumed would be done here?)
@pnkfelix well, it was `i32`, but I think we've regressed overnight, you might want to check that out.

**Pull request merged**